src/tcache.c

```diff
@@ -169,6 +312,61 @@ tcache_gc_small(tsd_t *tsd, tcache_slow_t *tcache_slow, tcache_t *tcache,
 	     tcache_slow->lg_fill_div[szind]) > 1) {
 		tcache_slow->lg_fill_div[szind]++;
 	}
+
+	/*
+	 * When the new tcache gc is not enabled, or simply the entire bin needs
+	 * to be flushed, flush the bottom nflush items directly.
+	 */
+	if (!opt_experimental_tcache_gc || nflush == ncached) {
+		goto label_flush;
+	}
+
+	/* Query arena binshard to get heuristic locality info. */
+	void *addr = tcache_gc_small_heuristic_addr_get(tsd, tcache_slow, szind);
+	if (addr == NULL) {
```

Comment:
We try to take slabcur first, if not available, check the slabs_nonfull heap, if still empty, the addr returns NULL, meaning that the current shard has no partially used slabs.

Reply:
I see. Since the case is rare and the heuristic way used to identify the locality cannot be trivially applied, it will just be flushing as before, right? I assume this corner case won't affect the locality much as well since it should rarely be touched?

Reply:
Yes, if no locality info can be found, it will just flush the bottom nflush items as before.